File: Seminar/src/main/java/org/sopt/Seminar/global/common/exception/SoptControllerAdvice.java
package org.sopt.Seminar.global.common.exception;

import static org.springframework.http.HttpStatus.BAD_REQUEST;
import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
import static org.springframework.http.HttpStatus.NOT_FOUND;

import org.sopt.Seminar.global.common.dto.ErrorResponse;
import org.sopt.Seminar.global.common.exception.SoptException.InvalidRequestException;
import org.sopt.Seminar.global.common.exception.SoptException.NotFoundException;
import org.sopt.Seminar.global.common.exception.SoptException.ServerErrorException;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class SoptControllerAdvice {

    @ExceptionHandler(InvalidRequestException.class)
    public ResponseEntity<ErrorResponse> handleInvalidRequestException(InvalidRequestException ex) {
        return ResponseEntity.status(BAD_REQUEST).body(new ErrorResponse(ex));
    }

    @ExceptionHandler(NotFoundException.class)
    public ResponseEntity<ErrorResponse> handleNotFoundException(NotFoundException ex) {
        return ResponseEntity.status(NOT_FOUND).body(new ErrorResponse(ex));
    }

    @ExceptionHandler(ServerErrorException.class)
    public ResponseEntity<ErrorResponse> handleServerErrorException(ServerErrorException ex) {
        return ResponseEntity.status(INTERNAL_SERVER_ERROR).body(new ErrorResponse(ex));
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/global/common/exception/SoptControllerAdvice.kt
package org.sopt.kotlinsemina.global.common.exception

import org.sopt.kotlinsemina.global.common.dto.ErrorResponse
import org.sopt.kotlinsemina.global.common.exception.SoptException.*
import org.springframework.http.HttpStatus
import org.springframework.http.HttpStatus.*
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class SoptControllerAdvice {
    @ExceptionHandler(InvalidRequestException::class)
    fun handleInvalidRequestException(ex: InvalidRequestException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.status(BAD_REQUEST).body(ErrorResponse(ex))
    }

    @ExceptionHandler(NotFoundException::class)
    fun handleNotFoundException(ex: NotFoundException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.status(NOT_FOUND).body(ErrorResponse(ex))
    }

    @ExceptionHandler(ServerErrorException::class)
    fun handleServerErrorException(ex: ServerErrorException): ResponseEntity<ErrorResponse> {
        return ResponseEntity.status(INTERNAL_SERVER_ERROR).body(ErrorResponse(ex))
    }
}